pkg/policy/groups/actions.go

```diff
@@ -135,24 +135,30 @@ func addDerivativeCNP(cnp *cilium_v2.CiliumNetworkPolicy) error {
 	})
 
 	var derivativeCNP *cilium_v2.CiliumNetworkPolicy
-	var err error
+	var derivativeErr error
+
+	// The maxNumberOfAttempts is to not hit the limits of cloud providers API.
+	// Also, the derivativeErr is never returned, if not the controller will
+	// hit this function and the cloud providers limit will be raised. This
+	// will cause a disaster, due all other policies will hit the limit as
+	// well.
+	// If the createDerivativeCNP() fails, a new all block rule will be inserted and
+	// the derivative status in the parent policy  will be updated with the
+	// error.
 	for numAttempts := 0; numAttempts <= maxNumberOfAttempts; numAttempts++ {
-		if numAttempts == maxNumberOfAttempts {
-			return fmt.Errorf("Cannot create derivative CNP due failures: %s", err)
-		}
-		derivativeCNP, err = createDerivativeCNP(cnp)
-		if err == nil {
+		derivativeCNP, derivativeErr = createDerivativeCNP(cnp)
+		if derivativeErr == nil {
 			break
 		}
-		scopedLog.WithError(err).Error("Cannot create derivative")
-		statusErr := updateDerivativeStatus(cnp, derivativeCNP.ObjectMeta.Name, err)
+		scopedLog.WithError(derivativeErr).Error("Cannot create derivative rule. Installing block all rule.")
+		statusErr := updateDerivativeStatus(cnp, derivativeCNP.ObjectMeta.Name, derivativeErr)
 		if statusErr != nil {
-			log.WithError(err).Error("Cannot update CNP status on invalid derivative")
+			scopedLog.WithError(statusErr).Error("Cannot update CNP status on invalid derivative")
```

Comment:
"cannot update CNP status for derivative policy"? "invalid derivative" is a bit vague.